dropshot/Cargo.toml

```diff
@@ -0,0 +1,38 @@
+[package]
+name = "dropshot"
+description = "expose REST APIs from a Rust program"
+version = "0.1.0"
+authors = ["David Pacheco <dap@oxidecomputer.com>"]
+edition = "2018"
+license = "Apache-2.0"
+repository = "https://github.com/oxidecomputer/dropshot/"
+
+[dependencies]
+async-trait = "0.1.24"
+bytes = "0.5.4"
+chrono = { version = "0.4", features = [ "serde" ] }
+dropshot_endpoint = { version = "0.1.0", path = "../dropshot_endpoint" }
+futures = "0.3.1"
+http = "0.2.0"
+hostname = "0.3.0"
+hyper = "0.13"
```

Comment:
For consistency's sake, IMHO we should either do

    "0.13"

or

    "0.13.0"

consistently across all deps. They're equivalent, so which one we choose doesn't matter much.

Reply:
Will look into this in a follow up (this was there before this change).

Reply:
It wasn't obvious to me that those would be equivalent, so I went looking for confirmation in a primary source.

[Cargo says it uses SemVer](https://doc.rust-lang.org/cargo/reference/manifest.html#the-version-field).  The [detailed docs on specifying dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html) describe semvers with carets, tildes, and wildcards -- none of which are part of the [SemVer spec](https://semver.org/), though they're obviously widely used in similar contexts (thanks to npm, I believe).  The [SemVer BNF](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions) says a semver *must* have a major, minor, and patch number -- you can't leave out the patch or minor numbers -- but as we said, the spec doesn't actually describe what can go in this field.

[The semver package that npm uses documents that in the specific case of hyphen ranges, you can leave off the trailing parts of the version number (it calls these "partial versions") and they'll be filled in with zeros. ](https://www.npmjs.com/package/semver#hyphen-ranges-xyz---abc)  I think there are cases where npm supports these partial versions that aren't documented there (e.g., as a plain package dependency, just like we're talking about here, not inside an explicit range).

I've obviously gone overboard here but I can't find anything that says these are the same.  I'm going to prefer the full written out version for consistency.

Reply:
I'll move my reply to #3 :)